gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ProcedureRequestMapper.java

```diff
@@ -0,0 +1,172 @@
+                .setValue(id);
+        return identifier;
+    }
+
+    private Annotation getNote(String text) {
+        if (StringUtils.isNotEmpty(text)) {
+            var note = new Annotation();
+            return note.setText(text);
+        }
+
+        return null;
+    }
+
+    private Date getAuthoredOn(TS availabilityTime, RCMRMT030101UK04EhrExtract ehrExtract, II planStatementID) {
+        if (availabilityTime != null) {
+            return DateFormatUtil.parse(availabilityTime.getValue()).getValue();
+        } else {
+            var ehrComposition = EhrResourceExtractorUtil.extractEhrCompositionForPlanStatement(ehrExtract, planStatementID);
+            if (ehrComposition.getAvailabilityTime() != null) {
+                return DateFormatUtil.parse(ehrComposition.getAvailabilityTime().getValue()).getValue();
+            } else if (ehrExtract.getAvailabilityTime() != null) {
+                return DateFormatUtil.parse(ehrExtract.getAvailabilityTime().getValue()).getValue();
+            }
+        }
+
+        return null;
+    }
+
+    private DateTimeType getOccurrenceDate(IVLTS effectiveTime) {
+        if (effectiveTime != null && effectiveTime.getCenter() != null) {
+            return DateFormatUtil.parse(effectiveTime.getCenter().getValue());
+        }
+
+        return null;
+    }
+
+    private Reference getAgentReference(List<RCMRMT030101UK04Participant> participantList, RCMRMT030101UK04EhrExtract ehrExtract,
+                                        II planStatementID) {
+        Reference reference = new Reference();
+        var nonNullFlavorParticipants = participantList.stream()
+                .filter(participant -> isNotNullFlavour(participant))
+                .collect(Collectors.toList());
+
+        var pprfParticipants = getParticipantReference(nonNullFlavorParticipants, PPRF_PERFORMER);
+        if (pprfParticipants.isPresent()) {
+            return reference.setReference(pprfParticipants.get());
+        }
+
+        var prfParticipants = getParticipantReference(nonNullFlavorParticipants, PRF_PERFORMER);
+        if (prfParticipants.isPresent()) {
+            return reference.setReference(prfParticipants.get());
+        }
+
+        var ehrComposition = EhrResourceExtractorUtil.extractEhrCompositionForPlanStatement(ehrExtract, planStatementID);
+        var participant2Reference = getParticipant2Reference(ehrComposition);
+        if (participant2Reference.isPresent()) {
+            return reference.setReference(participant2Reference.get());
+        }
+
+        // TODO: if none of these are present, then we should reference an 'Unknown User' Practitioner (NIAD-2026)
+
+        return reference;
+    }
+
+    private Optional<String> getParticipantReference(List<RCMRMT030101UK04Participant> participantList, String typeCode) {
+        return participantList.stream()
+                .filter(participant -> hasTypeCode(participant, typeCode))
```

Comment:
I think one indent is enough here. The same for stream inside getParticipant2Reference method and createProcedureRequest method.